### PR TITLE
Add building CLI app in CI

### PR DESCRIPTION
**Pull request by asklar**
- [x] need to rebase once the nuget change goes in

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-xaml/pull/153)

### CODE REVIEW

File: .github/workflows/SetUpAppForNuget.ps1 [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+[xml]$EF = Get-Content .\windows\ExperimentalFeatures.props
+$node = $EF.Project.PropertyGroup.WinUI2xVersion
+if ($null -eq $node) {
+  $node=$EF.CreateElement('WinUI2xVersion', $ns)
+  $v=$EF.CreateTextNode($WinUIVersion.ToString())
+  $node.AppendChild($v)
+  $EF.Project.PropertyGroup.AppendChild($node)
+} else {
+  $EF.Project.PropertyGroup.WinUI2xVersion = $WinUIVersion.ToString()
```
Comment (NickGerleman) [inline .github/workflows/SetUpAppForNuget.ps1:25]: Why do we need to override winui version?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -41,18 +62,102 @@ jobs:
...
+      run: yarn install
+
+    - name: build TS
+      run: yarn build
+
+    - name: yarn link
+      run: yarn link
+      working-directory: package
+
+    - name: create 0.64 app
```
Comment (NickGerleman) [inline .github/workflows/main.yml:92]: 0.64 here, but 0.65 in package.json?

File: .github/workflows/SetUpAppForNuget.ps1 [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    $node = $EF.CreateElement('ReactNativeWindowsVersion', $ns)
+    $v = $EF.CreateTextNode($ReactNativeWindowsVersion)
+    $node.AppendChild($v)
+    $EF.Project.PropertyGroup.AppendChild($node)
+  } else {
+    # not using NuGet and also don't have the prop already
+  }
+} else {
+  if ($UseNuGet) {
+    $EF.Project.PropertyGroup.ReactNativeWindowsVersion = $ReactNativeWindowsVersion
```
[the remark this answers is not in the document]
Comment (NickGerleman) [reply .github/workflows/SetUpAppForNuget.ps1:60]: Oh wait this is the one that was added later... Was thinking of `_ReactNativeWindowsVersionCheckNugetVersion`. Still kinda curious what we're doing here with this though.

File: .github/workflows/SetUpAppForNuget.ps1 (hunk 1 above)
Comment (asklar) [reply .github/workflows/SetUpAppForNuget.ps1:25]: 0.64 shipped with WinUI 2.3
react-native-xaml uses controls added in 2.6.

**Review by chiaramooney — APPROVED** [review]
👍 looks good to me. Out of curiosity how come you are using v0.64 for your app?

**Comment by asklar** [comment]
@chiaramooney we have some customers on 0.64 right now so this is mainly to ensure we don't break them :)

**Pull request merged**